wallet/src/account/utxo_selector/mod.rs

```diff
@@ -0,0 +1,774 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BinaryHeap;
+
+use common::{
+    chain::{
+        tokens::{OutputValue, TokenData},
+        TxOutput, UtxoOutPoint,
+    },
+    primitives::{signed_amount::SignedAmount, Amount},
+};
+use crypto::random::{make_pseudo_rng, Rng, SliceRandom};
+
+const TOTAL_TRIES: u32 = 100_000;
+
+/// A group of UTXOs paid to the same output script.
+/// This helps reduce privacy leaks resulting from address reuse.
+#[derive(Clone)]
+pub struct OutputGroup {
+    /// The list of UTXOs contained in this output group.
+    outputs: Vec<(UtxoOutPoint, TxOutput)>,
+    /// the total amount of the outputs in this group
+    value: Amount,
+    /// The fee to spend these UTXOs at the effective feerate.
+    fee: Amount,
+    /// The fee to spend these UTXOs at the long term feerate.
+    long_term_fee: Amount,
+    /// Total weight of the UTXOs in this group.
+    /// Non-witness data + witness data * Factor
```

Comment:
We don't use any kind of weighting... this is segwit bitcoin stuff. We can replace this with a normal size, I believe.

Reply:
changed the comment, it is still 0 for now